src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs

```diff
@@ -122,344 +117,32 @@ protected override void EndProcessing()
-                    };
-                if (EnumsAsStrings)
-                {
-                    jsonSettings.Converters.Add(new StringEnumConverter());
-                }
-
-                if (!Compress)
-                {
-                    jsonSettings.Formatting = Formatting.Indented;
-                }
-
-                string output = JsonConvert.SerializeObject(preprocessedObject, jsonSettings);
-                WriteObject(output);
-            }
-        }
-
-        #endregion overrides
-
-        /// <summary>
-        /// Return an alternate representation of the specified object that serializes the same JSON, except
-        /// that properties that cannot be evaluated are treated as having the value null.
-        /// Primitive types are returned verbatim.  Aggregate types are processed recursively.
-        /// </summary>
-        /// <param name="obj">The object to be processed.</param>
-        /// <param name="depth">The current depth into the object graph.</param>
-        /// <returns>An object suitable for serializing to JSON.</returns>
-        private object ProcessValue(object obj, int depth)
-        {
-            if (Stopping)
-            {
-                throw new StoppingException();
-            }
-
-            PSObject pso = obj as PSObject;
-
-            if (pso != null)
-                obj = pso.BaseObject;
-
-            object rv = obj;
-            bool isPurePSObj = false;
-            bool isCustomObj = false;
-
-            if (obj == null
-                || DBNull.Value.Equals(obj)
-                || obj is string
-                || obj is char
-                || obj is bool
-                || obj is DateTime
-                || obj is DateTimeOffset
-                || obj is Guid
-                || obj is Uri
-                || obj is double
-                || obj is float
-                || obj is decimal)
-            {
-                rv = obj;
-            }
-            else if (obj is Newtonsoft.Json.Linq.JObject jObject)
-            {
-                rv = jObject.ToObject<Dictionary<object, object>>();
-            }
-            else
-            {
-                TypeInfo t = obj.GetType().GetTypeInfo();
-
-                if (t.IsPrimitive)
-                {
-                    rv = obj;
-                }
-                else if (t.IsEnum)
+                object objectToProcess = (_inputObjects.Count > 1 || AsArray) ? (_inputObjects.ToArray() as object) : _inputObjects[0];
+
+                var context = new JsonObject.ConvertToJsonContext(
+                    Depth,
+                    cancellationSource.Token,
+                    EscapeHandling,
+                    EnumsAsStrings.IsPresent,
+                    Compress.IsPresent,
+                    targetCmdlet: this);
+
+                string output = JsonObject.ConvertToJson(objectToProcess, in context);
+                if (output != null)
```

Comment:
I believe so. In case `ctrl+c` is signaled, `JsonObject.ConvertToJson` would return null, and I think we don't want to write that null to pipeline.
I know that when the pipeline is stopping, `WriteObject` will throw from [`MshCommandRuntime._WriteObjectSkipAllowCheck`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/MshCommandRuntime.cs#L2576), so technically it might be OK to remove this check. But I want to play safe here. What do you think?

Reply:
I see that new code keep previous behavior and my thoughts was "is it useful to return null?" like `$null | ConvertTo-Json | Should -Be $null`

Reply:
Well, if we want to change that behavior, we should change the code [here](https://github.com/PowerShell/PowerShell/blob/7df2a22ea185bd27e680a7857d16b3007379c0bf/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs#L107). `JsonObject.ConvertToJson` returns null only if `StopException` is thrown, and it has nothing to do with enabling `$null | ConvertTo-Json`.

Yeah, `$null | ConvertTo-Json` seems like a reasonable scenario, but the test should be `$null | ConvertTo-Json | Should -Be "null"`. It's better to discuss this in a separate issue. Can you open an issue?
```
PS:30> [Newtonsoft.Json.JsonConvert]::SerializeObject($null) -eq "null"
True
```

Reply:
> Can you open an issue?

I think @markekraus could comment. If he confirm that the scenario is useful we'll open new issue.

Reply:
You never want ConvertFrom-Json write `$null` to the pipeline. But looking at this, the only scenario that would do that is a stopping exception. I'm not sure if there is any edge case where JsonConvert.SerializeObejct() would return a null string object, nothing I have tried to create such a scenario works anyway. It may be safe to remove it, but, I think it's safer to leave it.

Reply:
`$null | ConvertTo-Json` is not supported today -- the passed-in `$null` is ignored. Whether or not we should change the current behavior to support it should be a separate issue.